src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			var objectUid = GetObjectUid(objectDefinition);
+
+			if (objectUid != null)
+			{
+				var candidateProperties = FindLocalizableProperties(objectDefinition.Type)
+					.Except(objectDefinition.Members.Select(m => m.Member.Name));
+				foreach (var prop in candidateProperties)
+				{
+					var localizedValue = BuildLocalizedResourceValue(null, prop, objectUid);
+					if (localizedValue != null)
+					{
+						writer.AppendLineInvariant("{0} = {1},", prop, localizedValue);
+					}
+				}
+			}
+		}
+
+		private void TryValidateContentPresenterBinding(IIndentedStringBuilder writer, XamlObjectDefinition objectDefinition, XamlMemberDefinition member)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			if (
+				FindType(objectDefinition.Type) == _contentPresenterSymbol
+				&& member.Member.Name == "Content"
+			)
+			{
+				var binding = member.Objects.FirstOrDefault(o => o.Type.Name == "Binding");
+
+				if (binding != null)
+				{
+					var hasRelativeSource = binding.Members
+						.Any(m =>
+							m.Member.Name == "RelativeSource"
+						// It can either be TemplatedParent or Self. In either cases, it does not use the inherited
+						// DataContext, which falls outside of the scenario we want to avoid.
+						);
+
+					if (!hasRelativeSource)
+					{
+						writer.AppendFormatInvariant(
+							"\n#error Using a non-template binding expression on Content " +
+							"will likely result in an undefined runtime behavior, as ContentPresenter.Content overrides " +
+							"the local value of ContentPresenter.DataContent. " +
+							"Use ContentControl instead if you really need a normal binding on Content.\n");
+					}
+				}
+			}
+		}
+
+		private void BuildUiAutomationId(IIndentedStringBuilder writer, string closureName, string uiAutomationId, XamlObjectDefinition parent)
+		{
+			TryAnnotateWithGeneratorSource(writer);
+			if (uiAutomationId.IsNullOrEmpty())
+			{
+				return;
+			}
+
+			writer.AppendLineInvariant("// UI automation id: {0}", uiAutomationId);
+
+			// ContentDescription and AccessibilityIdentifier are used by Xamarin.UITest (Test Cloud) to identify visual elements
+			if (IsAndroidView(parent.Type))
+			{
+				writer.AppendLineInvariant("{0}.ContentDescription = \"{1}\";", closureName, uiAutomationId);
+			};
+
+			if (IsIOSUIView(parent.Type))
+			{
+				writer.AppendLineInvariant("{0}.AccessibilityIdentifier = \"{1}\";", closureName, uiAutomationId);
+			}
+		}
+
+		private bool IsRelativePanelSiblingProperty(string name)
+		{
+			return name.Equals("Above") ||
+				name.Equals("Below") ||
+				name.Equals("LeftOf") ||
+				name.Equals("RightOf") ||
+				name.Equals("AlignBottomWith") ||
+				name.Equals("AlignHorizontalCenterWith") ||
+				name.Equals("AlignLeftWith") ||
+				name.Equals("AlignRightWith") ||
+				name.Equals("AlignTopWith") ||
+				name.Equals("AlignVerticalCenterWith");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this call to 'name.Equals' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)